orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/tasks/MonitorWebhookTask.java

```diff
@@ -0,0 +1,270 @@
+ * Copyright 2017 Schibsted ASA.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.webhook.tasks;
+
+import com.google.common.base.Strings;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.PathNotFoundException;
+import com.netflix.spinnaker.orca.api.pipeline.OverridableTimeoutRetryableTask;
+import com.netflix.spinnaker.orca.api.pipeline.TaskResult;
+import com.netflix.spinnaker.orca.api.pipeline.models.ExecutionStatus;
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
+import com.netflix.spinnaker.orca.webhook.config.WebhookProperties;
+import com.netflix.spinnaker.orca.webhook.pipeline.WebhookStage;
+import com.netflix.spinnaker.orca.webhook.service.WebhookService;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpStatusCodeException;
+
+@Slf4j
+@Component
+public class MonitorWebhookTask implements OverridableTimeoutRetryableTask {
+  private static final String JSON_PATH_NOT_FOUND_ERR_FMT =
+      "Unable to parse %s: JSON property '%s' not found in response body";
+
+  private final long backoffPeriod = TimeUnit.SECONDS.toMillis(1);
+  private final long timeout = TimeUnit.HOURS.toMillis(1);
+
+  private final WebhookService webhookService;
+  private final WebhookProperties webhookProperties;
+
+  @Autowired
+  public MonitorWebhookTask(WebhookService webhookService, WebhookProperties webhookProperties) {
+    this.webhookService = webhookService;
+    this.webhookProperties = webhookProperties;
+  }
+
+  @Override
+  public long getBackoffPeriod() {
+    return backoffPeriod;
+  }
+
+  @Override
+  public long getTimeout() {
+    return timeout;
+  }
+
+  @Override
+  public long getDynamicBackoffPeriod(StageExecution stage, Duration taskDuration) {
+    if (taskDuration.toMillis() > TimeUnit.MINUTES.toMillis(1)) {
```

Comment:
this is surprising behavior! (not suggesting you change it in the scope of this PR but a 1s poll interval by default seems a bit much)